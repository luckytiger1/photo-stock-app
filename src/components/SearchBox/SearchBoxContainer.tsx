import React from 'react';
import { connect } from 'react-redux';
import { setTerm } from '../../redux/actions/fetchImages';
import { resetPage } from '../../redux/actions/advancePage';
import './SearchBox.scss';
import SearchBox from './SearchBox';

interface SearchBoxContainerProps {
  handleSearch: (data: string) => void;
  resetPageCount: () => void;
}

const SearchBoxContainer: React.FC<SearchBoxContainerProps> = ({
  handleSearch,
  resetPageCount,
}) => {
  const handleEnterKey = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      resetPageCount();
      handleSearch((e.target as HTMLInputElement).value);
    }
  };
  return <SearchBox handleEnterKey={handleEnterKey} />;
};

const mapDispatchToProps = {
  handleSearch: setTerm,
  resetPageCount: resetPage,
};

export default connect(null, mapDispatchToProps)(SearchBoxContainer);
