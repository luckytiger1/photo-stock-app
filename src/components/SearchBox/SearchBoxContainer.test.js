import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import { SearchBoxContainer } from './SearchBoxContainer';

it('<SearchBoxContainer /> renders correctly', () => {
  expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
  const mockEnterKey = jest.fn();
  expect(
    shallow(<SearchBoxContainer handleEnterKey={mockEnterKey} />),
  ).toMatchSnapshot();
});
