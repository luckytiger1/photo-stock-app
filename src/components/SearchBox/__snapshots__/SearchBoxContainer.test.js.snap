// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBoxContainer /> renders correctly 1`] = `
<SearchBox
  handleEnterKey={[Function]}
/>
`;
